File: agent-driven-architecture-main/examples/typescript/src/spine/ports/clock.ts
// ── spine/ports/clock — the only source of `now` (G9) ───────────────────────
// INTERFACES ONLY. A file in this folder with a body is a gate failure (C11).
// "A port is a published contract, not an implementation" (7.9/G13) is a
// property of the FOLDER here, not a convention someone remembers.

import type { Timestamp } from "../pure/ids";

export interface Clock {
  now(): Timestamp;
}

File: agent-driven-architecture-main/examples/kotlin/spine/src/main/kotlin/adr/spine/ports/Clock.kt
// ── spine/ports/clock — the time seam ──────────────────────────────────────
// A file in spine/ports/ with a body is a gate failure (check C11). A port is a
// published contract, not an implementation (7.9/G13) — here that is a property of
// the FOLDER, not a convention.

package adr.spine.ports

import adr.spine.pure.Timestamp

/** Read exactly once per step, by the boundary, and committed on the StepRecord (G9). */
interface Clock {
    fun now(): Timestamp
}
